tasks/internal/update-fbc-catalog-task/tests/test-update-fbc-catalog-default.yaml

```diff
@@ -0,0 +1,101 @@
+---
+apiVersion: tekton.dev/v1
+kind: Pipeline
+metadata:
+  name: test-update-fbc-catalog-default
+spec:
+  description: Test
+  tasks:
+    - name: run-task
+      taskRef:
+        name: update-fbc-catalog-task
+      params:
+        - name: fbcFragment
+          value: "registry.io/image0@sha256:0000"
+        - name: fromIndex
+          value: "quay.io/scoheb/fbc-index-testing:latest"
+        - name: targetIndex
+          value: "quay.io/fbc/catalog:test"
+        - name: buildTags
+          value: "[]"
+        - name: addArches
+          value: "[]"
+        - name: iibServiceAccountSecret
+          value: "iib-service-account-secret"
+        - name: subdirectory
+          value: $(context.pipelineRun.uid)
+    - name: check-result
+      params:
+        - name: jsonBuildInfo
+          value: $(tasks.run-task.results.jsonBuildInfo)
+        - name: buildState
+          value: $(tasks.run-task.results.buildState)
+        - name: genericResult
+          value: $(tasks.run-task.results.genericResult)
+        - name: indexImageDigests 
+          value: $(tasks.run-task.results.indexImageDigests)
+        - name: iibLog
+          value: $(tasks.run-task.results.iibLog)
+        - name: exitCode
+          value: $(tasks.run-task.results.exitCode)
+      taskSpec:
+        params:
+          - name: jsonBuildInfo
+            type: string
+          - name: buildState
+            type: string
+          - name: genericResult
+            type: string
+          - name: indexImageDigests
+            type: string
+          - name: iibLog
+            type: string
+          - name: exitCode
+            type: string
+        steps:
+          - name: check-result
+            image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+            script: |
+              #!/bin/bash
+              set -x
+
+              # the jsonBuild mockes has 9 keys on it
+              keyLength=$(jq '. | length' <<< '$(params.jsonBuildInfo)')
+              if [ "$keyLength" -ne 9 ]; then
+                echo "The task did not save a valid json in jsonBuildInfo result"
+                exit 1
+              fi
+
+              # shellcheck disable=SC2046
```

Comment:
what is this shellcheck warning wanting you to do?

Reply:
it is for the "Quote this to prevent word splitting." error. Normally adding the quotes would just fix the warning, but for some reason using it in the tekton parameters with a json content, it adds some extra single quotes that breaks jq. I tried some other ways to not require the double quotes, but they were always causing another error/warning, so ignoring the error looked better. I improved the comment.

Reply:
Got it. I think you can get around that issue by using the `env` directive like https://github.com/konflux-ci/release-service-catalog/blob/9666883410181fdad34e660b31ae921fe58c499b/tasks/managed/reduce-snapshot/reduce-snapshot.yaml#L36 but I am fine with just disabling shellcheck here

Reply:
@mmalina I will add the changes in the following up PR that will contain fixes/test for this task. Are you ok with that?